pkg/test/store/postgres/test_container.go

```diff
@@ -0,0 +1,110 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/kumahq/kuma/pkg/config"
+	pg_config "github.com/kumahq/kuma/pkg/config/plugins/resources/postgres"
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/plugins/common/postgres"
+)
+
+type PostgresContainer struct {
+	container testcontainers.Container
+	WithSsl   bool
+}
+
+func resourceDir() string {
+	dir, _ := os.Getwd()
+	for path.Base(dir) != "pkg" && path.Base(dir) != "app" {
+		dir = path.Dir(dir)
+	}
+	rDir := path.Join(path.Dir(dir), "pkg/test/store/postgres/resources")
+	err := os.Chmod(path.Join(rDir, "certs/postgres.client.key"), 0600)
+	if err != nil {
+		panic(err)
+	}
+	return rDir
+}
+
+func (v *PostgresContainer) Start() error {
+	ctx := context.Background()
+	req := testcontainers.ContainerRequest{
+		FromDockerfile: testcontainers.FromDockerfile{
+			Context: resourceDir(),
+		},
+		Env: map[string]string{
+			"POSTGRES_USER":     "kuma",
+			"POSTGRES_PASSWORD": "kuma",
+			"POSTGRES_DB":       "kuma",
+		},
+		ExposedPorts: []string{"5432/tcp"},
+		WaitingFor:   wait.ForListeningPort("5432"),
+	}
+	if v.WithSsl {
+		req.Cmd = strings.Split("-c ssl=on -c ssl_cert_file=/var/lib/postgresql/postgres.server.crt -c ssl_key_file=/var/lib/postgresql/postgres.server.key -c ssl_ca_file=/var/lib/postgresql/rootCA.crt -c hba_file=/var/lib/postgresql/pg_hba.conf", " ")
+	}
+	c, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		return err
+	}
+	v.container = c
+	return nil
+}
+
+func (v *PostgresContainer) Stop() error {
+	if v.container != nil {
+		return v.container.Terminate(context.Background())
+	}
+	return nil
+}
+
+func (v *PostgresContainer) Config(withRandomDB bool) (*pg_config.PostgresStoreConfig, error) {
```

Comment:
Suggest using constants `WithRandomDB` and `WithoutRandomDB` to make the call sites more readable.